src/NUnitFramework/framework/Attributes/CsvDataSourceAttribute.cs

```diff
@@ -0,0 +1,104 @@
+﻿// ***********************************************************************
+// Copyright (c) 2008-2015 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Collections;
+using System.Reflection;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Builders;
+
+
+namespace NUnit.Framework.Attributes
+{
+    /// <summary>
+    /// Imports CSV data from a file
+    /// </summary>
+    public class CsvDataAttribute : TestCaseSourceAttribute
+    {
+        private CsvData csv;
+
+        /// <summary>
+        /// Number of rows to read
+        /// </summary>
+        public int RowsToRead
+        {
+            get { return csv.RowsToRead; }
+            set { csv.RowsToRead = value; }
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="source"></param>
```

Comment:
I assume that source is the filename, but we are going to need much more complete XML Docs for public facing classes. That goes for all of your doc comments.